sv_inline.h

```diff
@@ -928,6 +928,34 @@ Perl_SvPV_helper(pTHX_
     return non_trivial(aTHX_ sv, lp, (flags|return_flags));
 }
 
+/*
+=for apidoc newRV_noinc
+
+Creates an RV wrapper for an SV.  The reference count for the original
+SV is B<not> incremented.
+
+=cut
+*/
+
+PERL_STATIC_INLINE SV *
+Perl_newRV_noinc(pTHX_ SV *const tmpRef)
+{
+    SV *sv = newSV_type(SVt_IV);
+
+    PERL_ARGS_ASSERT_NEWRV_NOINC;
+
+    SvTEMP_off(tmpRef);
+
+    /* inlined, simplified sv_setrv_noinc(sv, tmpRef); */
+    assert(!SvTHINKFIRST(sv));
```

Comment:
I don't understand why we would keep these asserts here.

Reply:
I was in two minds about not having them. Left them in as sort of a visual aid to the reader to show why the simplification was possible, but granted it's not that hard to figure out. Am happy to remove them.

Reply:
I've removed the asserts.